src/java.management/share/classes/java/lang/management/ThreadMXBean.java

```diff
@@ -637,6 +651,8 @@ public interface ThreadMXBean extends PlatformManagedObject {
      * <p>
      * This method is designed for troubleshooting use, but not for
      * synchronization control.  It might be an expensive operation.
+     * Its behavior with cycles involving virtual threads is not defined
+     * in this release.
```

Comment:
What does the current implementation return if the virtual threads are involved in a deadlock?    The class spec says "ThreadMXBean does not support monitoring or management of virtual threads" while this javadoc says it's undefined.    I wonder if it's helpful to include `@implNote` if appropriate.

Reply:
If there is cycle involving a virtual thread then the thread id of its carrier will be included in array. I think we can do better and include the thread id of the mounted thread.